### PR TITLE
fix(core): set correct mimetype for asset protocol streams, closes #5203

### LINKED ISSUE #5203: [bug] audio tag using a asset resource file as the source, the asset server returns wrong content-type of text/html
### Describe the bug

audio tag using a asset resource file as the source, the asset server returns wrong content-type of text/html

### Reproduction

create project with audio files in bundle.resources create a filepath using something like 
```ts
const filePath = convertFileSrc(
	await normalize(
		await resolveResource('name_of_audio_file.mp3')
	)
)
```
use in audio tag `<audio controls><source src="{filePath}" type="audio/mpeg" /></audio>`

Screen Shot 2022-09-15 at 16 53 31

### Expected behavior

the content type to be audio/mpeg

### Platform and versions

Environment
  › OS: Mac OS 12.2.1 X64
  › Node.js: 16.15.0
  › npm: 8.8.0
  › pnpm: 7.0.0
  › yarn: Not installed!
  › rustup: 1.25.1
  › rustc: 1.63.0
  › cargo: 1.63.0
  › Rust toolchain: stable-aarch64-apple-darwin 

Packages
  › @tauri-apps/cli [NPM]: 1.1.0
  › @tauri-apps/api [NPM]: 1.1.0
  › tauri [RUST]: 1.1.0,
  › tauri-build [RUST]: 1.1.0,
  › tao [RUST]: 0.14.0,
  › wry [RUST]: 0.21.0,

App
  › build-type: bundle
  › CSP: default-src 'self'; img-src 'self' asset: https://asset.localhost; media-src 'self' asset: https://asset.localhost;
  › distDir: ../dist
  › devPath: http://localhost:3000/
  › framework: Svelte

App directory structure
  ├─ node_modules
  ├─ public
  ├─ src-tauri
  ├─ .git
  ├─ .vscode
  └─ src

### Stack trace

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by amrbashir**
### What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [X] No

### Checklist
- [X] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [X] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [X] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: core/tauri/src/manager.rs [hunk 1]
```diff
@@ -626,7 +626,21 @@ impl<R: Runtime> WindowManager<R> {
             response = response.header(k, v);
           }
 
-          let mime_type = MimeType::parse(&data, &path);
+          let mime_type = {
+            use once_cell::sync::Lazy;
+            static MIME_TYPES_CACHE: Lazy<Arc<Mutex<HashMap<String, String>>>> =
```
Comment (lucasfernog) [inline core/tauri/src/manager.rs]: I'd try moving this data structure to the Manager struct

**Review by lucasfernog — CHANGES_REQUESTED** [review]

File: core/tauri/src/manager.rs (hunk 1 above)
Comment (amrbashir) [reply core/tauri/src/manager.rs]: the `InnerWindowManager` struct?

Comment (lucasfernog) [reply core/tauri/src/manager.rs]: yeah (I need to cleanup that manager struct, the WindowManager has too many stuff lol)

Comment (amrbashir) [reply core/tauri/src/manager.rs]: I think I could just use the `StateManager` to manage it, no?

Comment (lucasfernog) [reply core/tauri/src/manager.rs]: yeah that would be ok too

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**